evaluator/builtin_string.go

```diff
@@ -186,6 +187,26 @@ func builtinReverse(args []types.Datum, _ context.Context) (d types.Datum, err e
 	}
 }
 
+// See: http://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_space
+func builtinSpace(args []types.Datum, _ context.Context) (d types.Datum, err error) {
+	x := args[0]
+	switch x.Kind() {
+	case types.KindNull:
+		return d, nil
+	default:
+		v, err := x.ToInt64()
```

Comment:
It is better to return error when err != nil. We use strict mode which will treat warning as error.
mysql> select space("abc");
+--------------+
| space("abc") |
+--------------+
|              |
+--------------+
1 row in set, 2 warnings (0.01 sec)